rcp/org.testeditor.rcp4.bootstrap/src/org/testeditor/rcp4/bootstrap/Activator.java

```diff
@@ -0,0 +1,65 @@
+/*******************************************************************************
+ * Copyright (c) 2012 - 2016 Signal Iduna Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ * 
+ * Contributors:
+ * Signal Iduna Corporation - initial API and implementation
+ * akquinet AG
+ * itemis AG
+ *******************************************************************************/
+package org.testeditor.rcp4.bootstrap;
+
+import java.net.BindException;
+import java.net.ServerSocket;
+
+import org.osgi.framework.BundleActivator;
+import org.osgi.framework.BundleContext;
+
+public class Activator implements BundleActivator {
+
+	private static final String OSGI_HTTP_PORT = "org.osgi.service.http.port";
+	private static BundleContext context;
+
+	static BundleContext getContext() {
+		return context;
+	}
+
+	/*
+	 * (non-Javadoc)
+	 * 
+	 * @see org.osgi.framework.BundleActivator#start(org.osgi.framework.
+	 * BundleContext)
+	 */
+	public void start(BundleContext bundleContext) throws Exception {
+		Activator.context = bundleContext;
+		if (System.getProperty(OSGI_HTTP_PORT) != null) {
+			int retryCounter = 0;
+			boolean portIsInUse = true;
+			while (portIsInUse && retryCounter < 10) {
+				try {
+					retryCounter++;
+					ServerSocket socket = new ServerSocket(Integer.parseInt(System.getProperty(OSGI_HTTP_PORT)));
+					portIsInUse = false;
+					socket.close();
+				} catch (BindException e) {
+					int newPort = Integer.parseInt(System.getProperty(OSGI_HTTP_PORT)) + retryCounter;
```

Comment:
This looks uncommon to me. Why not use `new ServerSocket(0)` to get a free port? See http://stackoverflow.com/questions/2675362/how-to-find-an-available-port

Then we can get rid of the loop.